drivers/service_upgrade.go

```diff
@@ -263,7 +264,7 @@ func wait(apiClient *client.RancherClient, service *client.Service) error {
 func IsValidTag(tag string) error {
 	match := regTag.FindAllString(tag, -1)
 	if len(match) == 0 || len(match[0]) > 128 || (len(match[0]) != len(tag)) {
-		return fmt.Errorf("Invalid tag %s, tag length must be < 128, must contain [a-zA-Z0-9.-_] characters only, cannot start with '.' or '-'", tag)
+		return fmt.Errorf("Invalid tag %s, tag length must be < 128, must contain [a-zA-Z0-9.-_*] characters only, cannot start with '.' or '-'", tag)
```

Comment:
@broesch we wanted to use tags that are allowed by docker as per this document https://docs.docker.com/engine/reference/commandline/tag/#extended-description
In this `*` isn't a valid character.

Reply:
@mrajashree @broesch how's this going? we were using webhooks to deploy our platform and we just found out that we can't do versioning of tags because of this :(